src/openms/include/OpenMS/METADATA/PeptideHit.h

```diff
@@ -233,7 +233,7 @@ namespace OpenMS
     PeptideHit(double score,
                UInt rank,
                Int charge,
-               const AASequence& sequence);
+               AASequence sequence);
```

Comment:
is there a move inside?

Reply:
Yes. It uses the default move constructor of AASequence.

Reply:
I am not sure whether it is really better to 

1. make a copy of the object and then 2. move the object into another object 

compared to 

3. construct an object with a const-ref

I think constructing with a const-ref should always be faster since (1) and (3) should take the same amount of time (copy object vs construct from object) and the first solution takes extra time to then move. Comments?

Reply:
depends on the use-case:
if we pass in rvalues, the copy will be elided, i.e. it only costs a move.
If we always pass in l-values, const-ref is better.

Reply:
True, I decided on that because: In 1) we "only" pay an additional move if the developer does not use an r-value. In 3) we cannot use r-values at all. Did I get that right?

Reply:
> In 3) we cannot use r-values at all (at least we dont gain from it). Did I get that right?

I think for 3 it does not help but also does not hurt to use r-values. So I think we should always be better off with (3) since its better for l-values but neither better or worse for r-values. Right?

Reply:
I just did two overloads now. It was done the same already with the setter, so why not in the constructor? It was actually used with r-values e.g. in MzIdentML.